tools/spectral/ipa/__tests__/IPA110CollectionsUsePaginatedSchema.test.js

```diff
@@ -0,0 +1,336 @@
+        },
+      },
+    },
+  },
+  NonPaginatedList: {
+    type: 'object',
+    properties: {
+      count: {
+        type: 'integer',
+      },
+      items: {
+        type: 'array',
+        items: {
+          $ref: '#/components/schemas/Resource',
+        },
+      },
+    },
+  },
+};
+
+testRule('xgen-IPA-110-collections-use-paginated-schema', [
+  {
+    name: 'valid schemas with Paginated prefix',
+    document: {
+      paths: {
+        '/resources': {
+          get: {
+            responses: {
+              200: {
+                content: {
+                  'application/json': {
+                    schema: {
+                      $ref: '#/components/schemas/PaginatedResourceList',
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/resources/{id}': {
+          get: {
+            responses: {
+              200: {
+                content: {
+                  'application/json': {
+                    schema: {
+                      $ref: '#/components/schemas/Resource',
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+      components: {
+        schemas: componentSchemas,
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'valid schemas without Paginated prefix but with correct structure',
+    document: {
+      paths: {
+        '/resources': {
+          get: {
+            responses: {
+              200: {
+                content: {
+                  'application/json': {
+                    schema: {
+                      $ref: '#/components/schemas/ResourceList',
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/resources/{id}': {
+          get: {},
+        },
+      },
+      components: {
+        schemas: componentSchemas,
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid list methods without correct structure',
+    document: {
+      paths: {
+        '/resources': {
+          get: {
+            responses: {
+              200: {
+                content: {
+                  'application/json': {
+                    schema: {
+                      $ref: '#/components/schemas/NonPaginatedList',
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/resources/{id}': {
+          get: {},
+        },
+      },
+      components: {
+        schemas: componentSchemas,
+      },
+    },
+    errors: [
+      {
+        code: 'xgen-IPA-110-collections-use-paginated-schema',
+        message:
+          'List methods must use a paginated response schema. The response should reference a schema with either a name starting with "Paginated" or contain both "totalCount" (integer) and "results" (array) fields.',
```

Comment:
```suggestion
          'List methods must use a paginated response schema. The response should reference a schema with a name starting with "Paginated" and contain both "totalCount" (integer) and "results" (array) fields.',
```
On the error message, I think we should have the proper guidance that it's not enough to just prefix the name with `Paginated`

Reply:
I was thinking if we force people to use Paginated prefix, but why not? It brings consistency across the naming. There are some cases extending `ApiListView`, but the name does not have `Paginated` prefix.

I think I can also update the IPA guideline to be more clear on how we decide if it is paginated with the same words here. Wdyt?

Reply:
Related to wojciechs comment, I have no strong opinion but we should either require Paginated prefix or not, but we should always check for results and total count regardless of the schema name